diff --git a/CMake/TryRunResults-ParaView3-bgl-xlc.cmake b/CMake/TryRunResults-ParaView3-bgl-xlc.cmake
index 32259de..5edb87c 100644
--- a/CMake/TryRunResults-ParaView3-bgl-xlc.cmake
+++ b/CMake/TryRunResults-ParaView3-bgl-xlc.cmake
@@ -167,5 +167,5 @@ SET( HDF5_PRINTF_LL_TEST_RUN__TRYRUN_OUTPUT
 # xlc has a pthread.h which can be included, but it's basically empty
 # anyway check_include_files(pthread.h) returns TRUE, so a more
 # sophisticated test would be required. Just force the correct result here.
-SET(HAVE_PTHREAD_H FALSE CACHE BOOL "No usable pthread.h" FORCE)
-SET(CMAKE_HAVE_PTHREAD_H FALSE CACHE BOOL "No usable pthread.h" FORCE)
+#SET(HAVE_PTHREAD_H FALSE CACHE BOOL "No usable pthread.h" FORCE)
+#SET(CMAKE_HAVE_PTHREAD_H FALSE CACHE BOOL "No usable pthread.h" FORCE)
